pandas/io/excel/_base.py

```diff
@@ -902,6 +913,12 @@ def __init__(
         self, path_or_buffer, engine=None, storage_options: StorageOptions = None
     ):
         if engine is None:
+            warnings.warn(
+                "The default argument engine=None is deprecated. "
+                "Specify the engine argument to suppress this warning.",
+                FutureWarning,
+                stacklevel=2,
```

Comment:
> For the future warning, stacklevel should be 2 for ExcelFile and 4 for pd.read_excel. I think 2 (and in general, the minimum) is correct here because that way there is an actual trace in all cases, but wasn't certain.

Given that `read_excel` is used much more, it might make sense to ensure it is correct for `read_excel`

Reply:
At the expense of having usage of ExcelFile directly have a stack trace that starts two levels too high (which includes the possibility of no stack trace)? IMO going too deep here is preferable, at least then the offending line is always visible.

Another solution would be using `inspect`:

```
import inspect
caller = inspect.stack()[1]
if (
    caller.filename.endswith("pandas/io/excel/_base.py")
    and caller.function == "read_excel"
):
    stacklevel = 4
else:
    stacklevel = 2
```

Reply:
if you want to do this is is fine, or ust turn off the check for stacklevel where needed. i don't think its worth jumping thru too many hoops.

Reply:
@rhshadrach as far as I know, warnings don't show a full stack trace, but only point to a single location. So whether this single location is wrong "higher" in the stack vs "lower" in the stack, I personally don't find the lower (deeper) one any more informative or less misleading .. 
The only advantage that the the deeper one has is that it will always show that the warning is coming from internal pandas code, so can give the hint this has to do with pandas functionality. But I think that goal can also be achieved by having a clear warning message (for example, include "pandas.read_excel" in the message)